tex/gregoriotex.lua

```diff
@@ -199,12 +213,48 @@ local function include_gabc_score(gabc_file)
         else
             log("using the file %s without recompiling, as %s hasn't changed since last compilation.", tex_file, gabc_file)
         end
+	if not test_gregoriotex_apiversion(tex_file) then
+	   log("Recompiling %s because %s does not match the current gregoriotex api version.", gabc_file, tex_file)
+	   compile_gabc(gabc_file, tex_file)
+	end
     else
         compile_gabc(gabc_file, tex_file)
     end
     tex.print(string.format("\\input %s", tex_file))
 end
 
+local function include_gtex_score(gtex_file)
+    local file_root = gtex_file:gsub("%.gtex+$","")
+    local gabc_file = gtex_file:gsub("%.gtex+$", ".gabc")
+
+    if not lfs.isfile(gtex_file) then
+       log("the file %s does not exist. Searching for a gabc file.", gtex_file)
+        if lfs.isfile(file_root .. '.gabc') then
+	    compile_gabc(gabc_file, gtex_file)
+	    tex.print(string.format("\\input %s", gtex_file))
+	    return
+	else
+	    err("The %s.gabc file does not exist.", file_root)
+	    return
+	end
+    end
+
+    local gtex_timestamp = lfs.attributes(gtex_file).modification
+    local gabc_timestamp = lfs.attributes(gabc_file).modification
+    if gtex_timestamp < gabc_timestamp then
+	gregoriotex.compile_gabc(gabc_file, gtex_file)
+    else
+	log("using the file %s without recompiling, as %s hasn't changed since last compilation.", gtex_file, gabc_file)
```

Comment:
This comment isn't entirely true, because you won't know if you're actually using the file until you've checked the api version.  Perhaps delete everything up to the comma?